bcipy/acquisition/datastream/lsl_server.py

```diff
@@ -22,39 +29,39 @@ class LslDataServer(StoppableThread):
 
     Parameters
     ----------
-        params : dict(channels: list(str), hz(int))
+        device_spec : DeviceSpec
             parameters used to configure the server. Should have at least
             a list of channels and the sample frequency.
-        generator : Object (see generator.py for options)
-            used to generate the data to be served. Ex. random_data.
+        generator : optional Generator (see generator.py for options)
+            used to generate the data to be served. Uses random_data_generator
+            by default.
         include_meta: bool, optional
             if True, writes metadata to the outlet stream.
         add_markers: bool, optional
             if True, creates a the marker channel and streams data to this
             channel at a fixed frequency.
+        marker_stream_name: str, optional
+            name of the sample marker stream
     """
 
-    def __init__(self, params, generator, include_meta=True,
-                 add_markers=False, name='TestStream'):
+    def __init__(self, device_spec: DeviceSpec, generator: Generator = None, include_meta: bool = True,
+                 add_markers: bool = False, marker_stream_name: str = MARKER_STREAM_NAME):
         super(LslDataServer, self).__init__()
 
-        channels = params['channels']
-        self.sample_hz = int(params['hz'])
-        assert channels, "Channels must be provided as a parameter"
-
-        stream_name = params.get('name', name)
-        self.generator = generator
+        self.device_spec = device_spec
+        self.generator = generator or random_data_generator(channel_count=device_spec.channel_count)
 
-        log.debug("Starting server with params: %s", str(params))
-        info = StreamInfo(stream_name,
-                          "EEG", len(channels),
-                          self.sample_hz,
+        log.debug("Starting LSL server for device: %s", device_spec.name)
```

Comment:
use f strings here

Reply:
Logging is actually the one place where python is optimized to use the old string formatting style and f-strings can lead to some inefficiencies: https://docs.python.org/3/howto/logging.html#optimization. In this particular case it's probably not a big deal, but some code linters will provide a warning.

Reply:
I didn't know that! I just assumed f strings are more performant.